apps/aecore/src/aec_db.erl

```diff
@@ -508,11 +511,74 @@ find_headers_and_hash_at_height(Height) when is_integer(Height), Height >= 0 ->
 %% When benchmarked on an cloud SSD it is faster then mnesia:index_read followed by filter
 -spec find_key_headers_and_hash_at_height(pos_integer()) -> [{binary(), aec_headers:key_header()}].
 find_key_headers_and_hash_at_height(Height) when is_integer(Height), Height >= 0 ->
-    R = mnesia:dirty_select(aec_headers, [{ #aec_headers{key = '_', value = '$1', height = Height}
-                                          , [{'=:=', {element, 1, '$1'}, key_header}]
-                                          , ['$_']
-                                          }]),
-    [{Hash, aec_headers:from_db_header(Header)} || #aec_headers{key = Hash, value = Header} <- R].
+    case persistent_term:get({?MODULE, chain_migration, key_headers}, done) of
+        in_progress ->
+            R = mnesia:dirty_select(aec_headers, [{ #aec_headers{key = '_',
+                                                                 value = '$1',
+                                                                 height = Height}
+                                                  , [{'=:=', {element, 1, '$1'}, key_header}]
+                                                  , ['$_']}]),
+            [{Hash, aec_headers:from_db_header(Header)}
+             || #aec_headers{key = Hash, value = Header} <- R];
+        done ->
+            case persistent_term:get(?BYPASS, nobypass) of
```

Comment:
UNFEFINED BEHAVIOUR!!!
iterate_upper_bound is a optimization hint - not an obligation. 
Iterating past iterate_upper_bound is UB and anything might happen

Reply:
really? ok. Will fill